File: 01_gpio_digital/led_blink_task.c
#include <wiringPi.h>
#include <stdio.h>
#define LED_RED 4
#define LED_YELLOW 23
#define LED_GREEN 15

int main (void)
{
  wiringPiSetupGpio();
  pinMode (LED_RED, OUTPUT);
  pinMode (LED_YELLOW, OUTPUT);
  pinMode (LED_GREEN, OUTPUT);

  digitalWrite (LED_RED, HIGH) ; 
  delay (2000);
  printf("led red on\n");

  digitalWrite (LED_RED,  LOW) ; 
  delay (2000);
  printf("led red off\n");

  digitalWrite (LED_YELLOW, HIGH) ; 
  delay (2000);
  printf("led yellow on\n");

  digitalWrite (LE
  printf("led green off\n");

  pinMode (LED_RED, INPUT);
  pinMode (LED_YELLOW, INPUT);
  pinMode (LED_GREEN, INPUT);
  
  return 0;
}

File: 01_gpio_digital/led_blink_task.py
import RPi.GPIO as GPIO
import time

LED_RED = 7
LED_YELLOW = 8
LED_GREEN = 9
GPIO.setmode(GPIO.BCM)
GPIO.setup(LED_RED, GPIO.OUT)
GPIO.setup(LED_YELLOW, GPIO.OUT)
GPIO.setup(LED_GREEN, GPIO.OUT)

GPIO.output(LED_RED, GPIO.HIGH)
print("led red on")
time.sleep(2)
GPIO.output(LED_RED, GPIO.LOW)
print("led red off")
GPIO.output(LED_YELLOW, GPIO.HIGH)
print("led yellow on")
time.sleep(2)
GPIO.output(LED_YELLOW, GPIO.LOW)
print("led yellow off")
GPIO.output(LED_GREEN, GPIO.HIGH)
print("led green on")
time.sleep(2)
GPIO.output(LED_GREEN, GPIO.LOW)
print("led green off")

GPIO.cleanup()